spec/integration/strong_resource_spec.rb

```diff
@@ -132,21 +132,87 @@
           params[:person][:company_attributes] = {
             title: 'walmart',
             state_attributes: {
-              acronym: 'ny'
-            }
+            acronym: 'ny'
+          }
           }
         end
 
         it 'allows nesting relations' do
           expect(controller.strong_resource.to_h)
-            .to eq('name' => 'John', 'company_attributes' => {
-              'title' => 'walmart',
-              'state_attributes' => {
-                'acronym' => 'ny'
-              }
-            })
+          .to eq('name' => 'John', 'company_attributes' => {
+            'title' => 'walmart',
+            'state_attributes' => {
+            'acronym' => 'ny'
+          }
+          })
         end
       end
     end
   end
 end
+
+describe "strong_resources" do
+  context "with multiple definitions" do
+    let(:controller) do
+      controller = CompaniesController.new
+      controller.params = ActionController::Parameters.new(params)
+      allow(controller).to receive(:action_name) { :create }
+      controller
+    end
+
+
+    context "request payload strong_resource == 'parent_company'" do
+      let(:params) do
+        { parent_company: { title: 'John' }, data: { type: 'parent_company' } }
+      end
+
+      it 'core params are allowed' do
+        params[:parent_company][:foo] = 'bar'
+        expect(controller.strong_resource.to_h).to eq('title' => 'John')
+      end
+
+      it 'relation params are allowed' do
+        params[:parent_company][:parent_company_attributes] =  {
+          title: 'Parent Company'
+        }
+        params[:parent_company][:state_attributes] =  {
+          acronym: 'VA'
+        }
+        expect(controller.strong_resource.to_h).to eq('title' => 'John', 'state_attributes' => { 'acronym' => 'VA' })
+      end
+    end
+
+    context "request payload strong_resource == 'company'" do
+      let(:params) do
+        { company: { title: 'John' }, data: { type: 'company' } }
+      end
+
+      it 'core params are allowed' do
+        params[:company][:foo] = 'bar'
+        expect(controller.strong_resource.to_h).to eq('title' => 'John')
+      end
+
+      it 'relation params are allowed' do
+        params[:company][:parent_company_attributes] =  {
+          title: 'Parent Company'
+        }
+        params[:company][:state_attributes] =  {
+          acronym: 'VA'
+        }
+        expect(controller.strong_resource.to_h).to eq('title' => 'John', 
+                                                      'state_attributes' => { 'acronym' => 'VA' },
+                                                      'parent_company_attributes' => { 'title' => 'Parent Company' })
+      end
+    end
+
+    context "request payload strong_resource == 'unicorn'" do
```

Comment:
If the premise of this test is handling strong resources without a block, let's make that the test description